src/EditorFeatures/CSharp/Formatting/Indentation/CSharpIndentationService.Indenter.cs

```diff
@@ -251,6 +251,17 @@ private IndentationResult GetIndentationBasedOnToken(SyntaxToken token)
                             return GetIndentationFromCommaSeparatedList(token);
                         }
 
+                    case SyntaxKind.CloseParenToken:
+                        {
```

Comment:
Maybe, this should include the check for null, like for the `SyntaxKind.CloseBracketToken` case. However, I don't see this check being used for every `case`, so I don't know if it's needed.

````csharp
var nonTerminalNode = token.Parent;
Contract.ThrowIfNull(nonTerminalNode, @"Malformed code or bug in parser???");
````

Reply:
The `IsKind` extension method includes a null check.

https://github.com/dotnet/roslyn/blob/af832ba32bdf95799f131459db1a0025ed9858aa/src/Compilers/CSharp/Portable/CSharpExtensions.cs#L27-L30

Reply:
Ah, I see. The `Contract.ThrowIfNull` is only being used in call sites where `IsKind` is not being used. 👍